tests/test_dnssec.py

```diff
@@ -880,6 +934,46 @@ def testToTimestamp(self):
         ts = dns.dnssec.to_timestamp(441812220)
         self.assertEqual(ts, REFERENCE_TIMESTAMP)
 
+    def test_sign_zone(self):
+        zone = dns.zone.from_text(test_zone_sans_nsec, "example.", relativize=False)
+
+        private_key = ed25519.Ed25519PrivateKey.generate()
+        algorithm = dns.dnssec.Algorithm.ED25519
+        dnskey = dns.dnssec.make_dnskey(
+            public_key=private_key.public_key(), algorithm=algorithm
+        )
+        lifetime = 3600
+
+        with zone.writer() as txn:
+            dns.dnssec.sign_zone(
+                zone=zone,
+                txn=txn,
+                keys=[(private_key, dnskey)],
+                lifetime=lifetime,
+            )
+
+        rrsigs = set(
+            [
+                (str(name), rdataset.covers)
+                for (name, rdataset) in zone.iterate_rdatasets()
+                if rdataset.rdtype == dns.rdatatype.RRSIG
+            ]
+        )
+        self.assertEqual(rrsigs, test_zone_rrsigs)
```

Comment:
Should there be some checking that the right RRSIGs are added?  And that both the zsk/ksk and csk versions do the expected thing for the dnskey set?

Reply:
Fixed in 054070b3c4c5963901b9c695461a99e4e1818584